plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -625,6 +634,58 @@ def finalize(self):
                 add_metadata_file_metadata('updateinfo', self.context.metadata_file_path,
                                            self.context.checksum)
 
+    def _get_pkglist_to_publish(self, erratum_unit):
+        """
+        Make a list of packages which will be listed in the published erratum.
+
+        It is possible to filter packages only if `repo_unit_nevra` is available,
+        otherwise publish everything except duplicated collections.
+
+        :param erratum_unit: the erratum unit to analyze
+        :param type: pulp_rpm.plugins.db.models.Errata
+        :return: list of pkglists to publish
+        :rtype: list of dicts
+        """
+        pkglists_to_publish = []
+
+        # If we can pull a repo_id off the conduit, use that to generate repo-specific nevra
+        if self.context.conduit and hasattr(self.context.conduit, 'repo_id'):
+            repo_unit_nevra = self.context.get_repo_unit_nevra(erratum_unit)
+        else:
+            repo_unit_nevra = None
```

Comment:
This goes on to do extra work that is then thrown out for each item in the loop below. I think it would be good to move this conditional up to the top of the method, and if `repo_unit_nevra is None`, `return erratum_unit.pkglist` to short out before doing any work.